voyager-transitions/src/commonMain/kotlin/cafe/adriel/voyager/transitions/ScreenTransition.kt

```diff
@@ -42,7 +64,30 @@ public fun ScreenTransition(
 ) {
     AnimatedContent(
         targetState = navigator.lastItem,
-        transitionSpec = transition,
+        transitionSpec = {
+            val contentTransform = transition()
+
+            val isPop = navigator.lastEvent == StackEvent.Pop
+
+            val screenEnterTransition = if (isPop) {
+                (targetState as? ScreenTransition)?.enter()
```

Comment:
Something strange here. Same code called when isPop and !isPop. I think, we should pass isPop parameter to enter function to define direction of entering, for example in slide transition